Mathlib/LinearAlgebra/Contraction.lean

```diff
@@ -117,6 +126,21 @@ theorem toMatrix_dualTensorHom {m : Type*} {n : Type*} [Fintype m] [Finite n] [D
   rw [and_iff_not_or_not, Classical.not_not] at hij
   rcases hij with hij | hij <;> simp [hij]
 
+/-- If the identity linear map lies in the range of the canonical map `M* ⊗[R] M → Hom_R(M, M)`,
+then `M` is a finite projective `R`-module. -/
+theorem finite_projective_of_one_mem_range_dualTensorHom (h : 1 ∈ range (dualTensorHom R M M)) :
+    Module.Finite R M ∧ Projective R M := by
```

Comment:
I think this would make more sense as two separate lemmas, unless there's precedent for this combination specifically.

Reply:
@eric-wieser "finite and projective" is now the standard idiom to refer to the condition we call "locally free" in literature which unfortunately does not exist in Mathlib. This is not a comment for or against the splitting, I just thought I would add more context.